version: "3.9"

volumes:
  postgres_data: {}

services:
  postgres:
    image: postgres
    volumes:
      - postgres_data:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=postgres
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    ports:
      - "5432:5432"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -q -d postgres -U postgres"]
      timeout: 10s
      interval: 5s
      retries: 15
  django:
    image: django
    volumes:
      - .:/chaoticbook
    ports:
      - "8000:8000"
    command:
      python manage.py runserver 0.0.0.0:8000
    depends_on:
      postgres:
        condition: service_healthy
